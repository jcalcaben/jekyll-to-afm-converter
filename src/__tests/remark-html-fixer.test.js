const { TestScheduler } = require("jest");
var unified = require("unified");
var markdown = require("remark-parse");

const fixer = require("../remark-html-fixer");

test("closes open tags", () => {
  const tree = unified().use(markdown).parse(`**Hello** 
    <br> 
    _world_`);

  unified()
    .use(fixer)
    .run(tree, (err, result) => {
      expect(result).toMatchSnapshot();
    });
});
