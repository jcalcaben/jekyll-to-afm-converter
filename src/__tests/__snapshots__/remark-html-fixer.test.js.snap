// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`closes open tags 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "position": Object {
                "end": Object {
                  "column": 8,
                  "line": 1,
                  "offset": 7,
                },
                "start": Object {
                  "column": 3,
                  "line": 1,
                  "offset": 2,
                },
              },
              "type": "text",
              "value": "Hello",
            },
          ],
          "position": Object {
            "end": Object {
              "column": 10,
              "line": 1,
              "offset": 9,
            },
            "start": Object {
              "column": 1,
              "line": 1,
              "offset": 0,
            },
          },
          "type": "strong",
        },
        Object {
          "position": Object {
            "end": Object {
              "column": 1,
              "line": 2,
              "offset": 11,
            },
            "start": Object {
              "column": 11,
              "line": 1,
              "offset": 10,
            },
          },
          "type": "text",
          "value": "
",
        },
        Object {
          "position": Object {
            "end": Object {
              "column": 9,
              "line": 2,
              "offset": 19,
            },
            "start": Object {
              "column": 5,
              "line": 2,
              "offset": 15,
            },
          },
          "type": "html",
          "value": "<br />",
        },
        Object {
          "position": Object {
            "end": Object {
              "column": 1,
              "line": 3,
              "offset": 21,
            },
            "start": Object {
              "column": 10,
              "line": 2,
              "offset": 20,
            },
          },
          "type": "text",
          "value": "
",
        },
        Object {
          "children": Array [
            Object {
              "position": Object {
                "end": Object {
                  "column": 11,
                  "line": 3,
                  "offset": 31,
                },
                "start": Object {
                  "column": 6,
                  "line": 3,
                  "offset": 26,
                },
              },
              "type": "text",
              "value": "world",
            },
          ],
          "position": Object {
            "end": Object {
              "column": 12,
              "line": 3,
              "offset": 32,
            },
            "start": Object {
              "column": 5,
              "line": 3,
              "offset": 25,
            },
          },
          "type": "emphasis",
        },
      ],
      "position": Object {
        "end": Object {
          "column": 12,
          "line": 3,
          "offset": 32,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 12,
      "line": 3,
      "offset": 32,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;
